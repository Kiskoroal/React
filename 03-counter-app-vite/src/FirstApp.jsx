import PropTypes from 'prop-types';
// const getSaludo = {
//     saludo: 'Hola, que tal estas don ',
//     nombre: 'Mamolo'
// };

// const newMessage = {
//     message: 'Hola Mundo',
//     title: 'Fernando',
// };
                                                                                                                       
export const FirstApp = ( {title, subtitle, name} ) => { 

    return(              
        <div data-testid = 'test-title'>
        <h2> { title } </h2>
        {/* <code>{ JSON.stringify(newMessage) }</code> */}
        <p>{ subtitle }</p>
        <p>{ subtitle }</p>
        <p>{ name }</p>
        </div>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
    name: PropTypes.string
}

FirstApp.defaultProps = {
    title: 'No hay ningún título',
    subtitle: 'No hay subtítulo',
    // name : 'No hay nombre'
}