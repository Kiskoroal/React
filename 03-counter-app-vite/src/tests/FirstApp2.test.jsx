/** @jest-environment jsdom */
import { render, screen } from "@testing-library/react";
import { FirstApp } from '../FirstApp'

describe('Pruebas en <FirstApp />', () => {
  
    const title = 'Hola, Soy Goku';
    const subtitle = 'Hola soy un subtítulo';

    test('Match con snapshot', () => {

        const { container } = render( <FirstApp title = {title} subtitle = {subtitle} /> )

        expect(container).toMatchSnapshot();
    })

    test('debe de mostrar el mensaje "Hola, soy Goku"', () => {
        render( <FirstApp title = {title}  subtitle = {subtitle} /> );

        expect( screen.getByText(title)).toBeTruthy();
    });

    test('debe de mostrar el titulo en un h1', () => {
        render(<FirstApp title={title} subtitle= {subtitle} />);
        expect(screen.getByRole('heading', {level: 2}).innerHTML ).toContain( title );
    })

    test('debe mostrar el título enviado por props', () => {
        
        render(<FirstApp title = { title } subtitle = { subtitle } />);
        expect(screen.getAllByText(subtitle).length ).toBe(2);
    })

})
