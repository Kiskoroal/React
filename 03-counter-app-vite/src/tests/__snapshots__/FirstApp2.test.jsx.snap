// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Match con snapshot 1`] = `
<div>
  <div
    data-testid="test-title"
  >
    <h2>
       
      Hola, Soy Goku
       
    </h2>
    <p>
      Hola soy un subtítulo
    </p>
    <p>
      Hola soy un subtítulo
    </p>
    <p />
  </div>
</div>
`;
